C/common/include/readingset_circularbuffer.h

```diff
@@ -33,7 +33,7 @@ class ReadingSetCircularBuffer {
 		
 
 	private:
-		std::mutex	m_mutex;
+		std::recursive_mutex	m_mutex;
```

Comment:
I am not sure if it is a good idea to use recursive_mutex everywhere and let mutex be acquired multiple times recursively even in a single use-case flow. It is probably better to just "check for buffer full and handle overflow" in append() and "check for buffer empty and return empty ReadingSet vector" in extract() in a single function and remove checkFull/checkEmpty functions.